### ISSUE TITLE
Cant call setState, forceUpdate on an unmounted Component

### ISSUE THREAD

**Issue by shelldandy** [jfilter/react-native-onboarding-swiper#20]
Using with `create-react-native-app`

What I'm doing is onDone and onSkip navigating to another screen:

```js
import React from 'react';
import { Image } from 'react-native'
import Onboarding from 'react-native-onboarding-swiper';

const OnboardingView = ({ navigation, ...props }) =>
  <Onboarding
    pages={[
      {
        backgroundColor: '#fff',
        image: <Image source={require('./circle.png')} />,
        title: 'Onboarding',
        subtitle: 'Done with React Native Onboarding Swiper',
      },
      {
        backgroundColor: '#fe6e58',
        image: <Image source={require('./square.png')} />,
        title: 'The Title',
        subtitle: 'This is the subtitle that sumplements the title.',
      },
      {
        backgroundColor: '#999',
        image: <Image source={require('./triangle.png')} />,
        title: 'Triangle',
        subtitle: "Beautiful, isn't it?",
      },
    ]}
    onDone={() => navigation.navigate('Terms')}
    onSkip={() => navigation.navigate('Terms')}
  />

export default OnboardingView
```

<img width="433" alt="screen shot 2018-04-30 at 3 18 54 pm" src="https://user-images.githubusercontent.com/7088511/39448189-cfe54752-4c89-11e8-9d43-412482d9ae3b.png">

**Comment by mike3run** [comment]
Of course, for navigation im using `react-navigation`

https://v2.reactnavigation.org/docs/en/getting-started.html

So basically a `navigator` prop is injected on all the views:

Here's a super minimal situation:

- Create-react-native-app let's go
- `yarn add react-navigation react-navigation`
- Also have the pngs from the simple example from here (circle, triangle, square)

`app.js`

```js
import React, { Component } from 'react';
import {View, Image, Text} from 'react-native'
import Onboarding from 'react-native-onboarding-swiper';
import { SwitchNavigator } from 'react-navigation';

class Splash extends Component {
   componentDidMount() {
    // Fake ajax/corroboration whatever
    setTimeout(() => {
      this.props.navigation.navigate('Onboarding')
    }, 3000)
  }

  render() {
    return (
      <View>
        <Image source={require('./circle.png')} />
      </View>
    );
  }
}


const OnboardingView = ({ navigation, ...props }) =>
  <Onboarding
    pages={[
      {
        backgroundColor: '#fff',
        image: <Image source={require('./circle.png')} />,
        title: 'Onboarding',
        subtitle: 'Done with React Native Onboarding Swiper',
      },
      {
        backgroundColor: '#fe6e58',
        image: <Image source={require('./square.png')} />,
        title: 'The Title',
        subtitle: 'This is the subtitle that sumplements the title.',
      },
      {
        backgroundColor: '#999',
        image: <Image source={require('./triangle.png')} />,
        title: 'Triangle',
        subtitle: "Beautiful, isn't it?",
      },
    ]}
    onDone={() => navigation.navigate('Another')}
    onSkip={() => navigation.navigate('Another')}
  />

const Another = () => <View><Text>Hello</Text></View>

export default SwitchNavigator({
  Splash: Splash,
  Onboarding: OnboardingView,
  Another: Another
})
```

**Comment by mike3run** [comment]
So in that example on loading you should get the Splash screen, after 2 seconds go to the onboarding and onDone or onSkip go to Another, that's when the error triggers.

I'll record a video...

**Comment by mike3run** [comment]
![swiper](https://user-images.githubusercontent.com/7088511/39452278-474fe100-4c97-11e8-9bd7-df8b30f55284.gif)

**Comment by peterchibunna** [comment]
I didn't use this component in a navigator yet I get the same warning when I navigate away.

**Comment by tornerdo** [comment]
It seems the problem originates [here](https://github.com/jfilter/react-native-onboarding-swiper/blob/e1697c737d8ef5c415cfe423a8b92502212b77b7/src/Pagination.js#L36) and [here](https://github.com/jfilter/react-native-onboarding-swiper/blob/e1697c737d8ef5c415cfe423a8b92502212b77b7/src/Pagination.js#L58).

A timeout of 500ms is set after either skip or finished is pressed while onSkip and onDone is called immediately. This leads to the situation where the timeout fires are the component is already unmounted.

Solutions would be to either to remove the timeout in the component or to fire the navigation event after more than 500ms, for example:
```javascript
    onDone={() => setTimeout(this.onFinish, 600)}
```

**Comment by jfilter** [comment]
Sorry guys, I didn't have time to look into this. @tornerdo is right.

This `setTimeout` was a hack and I could come up with a better solution during that time. It was needed to (re-)set the color of the status bar after leaving the onboarding screens: https://github.com/jfilter/react-native-onboarding-swiper/blob/master/src/index.js#L143

I am pretty sure that there is better way. If somebody find's a proper solution, please send a PR.

**Comment by jfilter** [comment]
fixed in #24
